src/SingleConnection.php

```diff
@@ -99,18 +113,33 @@ public function getDriverNamespace(): string
     /**
      * Connect.
      */
-    public function connect()
+    public function connect(?ConnectionOptions $options = null)
     {
         $this
             ->driver
             ->connect($this->credentials);
+
+        if ($options != null && $options->getKeepAliveIntervalSec() > 0) {
+            $this->keepAliveTimer = Loop::get()->addPeriodicTimer($options->getKeepAliveIntervalSec(), function() {
+                $qb = $this->createQueryBuilder();
+                $this->query(
+                    $qb
```

Comment:
Are we sure that this format works properly in all adapters? Would make sense to add a `ping()` method in the interface with all implementations and tested.

Reply:
for all current adapters postgresql, sqlite and mysql SELECT 1 is working. I verified mysql in our own implementation, and used online tools for the other 2:
https://extendsclass.com/postgresql-online.html
https://sqliteonline.com/

SELECT 1 is probably the smallest possible SELECT statement there is , so I cannot imagine any sql database not supporting this.

If you still have second thoughts let me know, I will write the ping but it will be equal for all current adapters